R/check_post_tags.R

```diff
@@ -0,0 +1,18 @@
+# Get list of blog posts ----
+posts <- list.files("posts", recursive = TRUE, pattern = "*.qmd")
+
+# Function to extract tags from a post and check them against the allowed list ----
+check_post_tags <- function(post) {
+  allowable_tags <- c("Metadata", "SDTM", "ADaM", "TLG", "Shiny", "Community", "Conferences", "Submissions", "Technical")
+
+  post_tags <- rmarkdown::yaml_front_matter(file.path("posts", post))$categories
+
+  cross_check <- post_tags %in% allowable_tags
+
+  if (!all(cross_check)) {
+    stop(paste("The tags selected in:", post, "are not all from the allowed list of tags."))
```

Comment:
You could implement the suggestion by @bms63 to list the problematic tags with something like this: 

```
problematic_tags <- post_tags[!cross_check]

message <- paste("The tags", enumerate(problematic_tags), " in:", post, "are not from the allowed list of tags.\n Please select from the following tags or contact one of the maintainers: ", enumerate(allowable_tags))

```